pkg/store/proxy.go

```diff
@@ -342,6 +348,12 @@ func (s *ProxyStore) Series(originalRequest *storepb.SeriesRequest, srv storepb.
 		}
 	}
 
+	if unsortedSeriesSetDetected || respHeap.MultiReplicaRespSetDetected() {
```

Comment:
```suggestion
	if unsortedSeriesSetDetected && respHeap.MultiReplicaRespSetDetected() {
```
Shouldn't it be AND?

Also are we sure we don't break compatibility with this error? I think we should just do extra sort if we have those stores ,no?

Reply:
Hm, we need to sort if either of those two cases is true, so I think this should be an or.

> Also are we sure we don't break compatibility with this error? I think we should just do extra sort if we have those stores ,no?

Yes we just do an extra sort in the querier when we see this warning. That's the only place where we have the possibility to sort when we use the lazy approach.

Reply:
I added an exception in the proxy to forward this error instead of aborting so that layered proxies can also propagate this information.

Reply:
I added an exception in the proxy to forward this error instead of aborting so that layered proxies can also propagate this information.